nixos/modules/services/databases/cockroachdb.nix

```
@@ -6,46 +6,46 @@ let
   cfg = config.services.cockroachdb;
   crdb = cfg.package;
 
-  escape    = builtins.replaceStrings ["%"] ["%%"];
-  ifNotNull = v: s: optionalString (v != null) s;
+  escape = builtins.replaceStrings [ "%" ] [ "%%" ];
 
-  startupCommand = lib.concatStringsSep " "
-    [ # Basic startup
-      "${crdb}/bin/cockroach start"
+  startupCommand = lib.escapeShellArgs
```

```suggestion
  startupCommand = utils.escapeSystemdExecArgs
```
this goes into an `ExecStart` directive, which does *not* obey shell syntax (only mostly!).

---

Thanks :) But the existing module has some existing ad-hoc escaping for systemd's `%` syntax. I'll need to remove that too.

---

Done. I squashed away the `escapeShellArgs` commit because it turned out to be wrong.